File: example/periphery/gpio/cpp/test_gpio_write.cc
#include <lockzhiner_vision_module/periphery/gpio/gpio.h>

#include <iostream>
#include <thread>

int main() {
  lockzhiner_vision_module::periphery::GPIO0A0 gpio;

  if (!gpio.Config(lockzhiner_vision_module::periphery::GPIOMode::OUT)) {
    std::cout << "Failed to config gpio mode" << std::endl;
    return 1;
  }

  if (!gpio.Write(lockzhiner_vision_module::periphery::GPIOState::HIGH)) {
    std::cout << "Failed to config gpio mode" << std::endl;
    return 1;
  }

  for (int i = 0; i < 10; i++) {
    std::cout << "Wait: " << i << "/" << 10 << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(1));
  }

  if (!gpio.Write(lockzhiner_vision_module::periphery::GPIOState::LOW)) {
    std::cout << "Failed to config gpio mode" << std::endl;
    return 1;
  }

  return 0;
}

File: example/periphery/gpio/python/test_gpio_write.py
from lockzhiner_vision_module.periphery import GPIO0A0, GPIOMode, GPIOState
import time

if __name__ == "__main__":
    gpio = GPIO0A0()

    if gpio.config(GPIOMode.OUT) is False:
        print("Failed to config gpio mode")
        exit(1)

    if gpio.write(GPIOState.HIGH) is False:
        print("Failed to write gpio state")
        exit(1)

    time_index = 0
    total_time = 10
    while time_index < total_time:
        print(f"Wait: {time_index}/{total_time}")
        time_index += 1
        time.sleep(1)

    if gpio.write(GPIOState.LOW) is False:
        print("Failed to write gpio state")
        exit(1)
